src/Adapter/DbSelect.php

```diff
@@ -150,4 +151,17 @@ protected function getSelectCount()
 
         return $countSelect;
     }
+
+    /**
+     * @internal
+     */
+    public function jsonSerialize()
+    {
+        return [
+            'select'       => $this->sql->buildSqlString($this->select),
+            'count_select' => $this->sql->buildSqlString(
+                $this->getSelectCount()
+            ),
+        ];
+    }
```

Comment:
I am a bit confused here, as coverage does not show usage of this method.
Is it an issue with the coverage or something is wrong with that change?

Reply:
Yes, it's a bit hidden: https://github.com/laminas/laminas-paginator/blob/b93f2e7d08ed548a95c1aa69d205e30da7f90eb3/src/Paginator.php#L853-L870

Reply:
I have also thought about marking the method as deprecated because version 3 of the paginator component should extract the entire caching and find a better solution for this _workaround_.